### ISSUE TITLE
How to fix do-expression from showing no-unused-expression message

### ISSUE THREAD

**Issue by ianizaguirre** [babel/eslint-plugin-babel#176]
**problem**: babel/no-unused-expressions: fails when using do expressions.

An example block that is producing the following message:
`Expected an assignment or function call and instead saw an expression.`

```js
const Header = () => {
  const test = true;
  return (
    <Wrapper>
      {do {
        if (test === 'randomExample') {
          <NavOne />;             // Message shows here
        } else if (test === true) {
          <NavTwo />;            // Message shows here
        }
      }}
    </Wrapper>
  );
};
```

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [@babel/eslint-plugin has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
